File: ex_lista/src_python/PontoFlutuante.py
'''
Ana Cristina Silva de Oliveira - 11965630
BCC - A ex06 Ponto flutuante
'''

def main():
    try:
        numero = float(input("Digite um valor decimal diferente de zero\n"))
        maior = numero
        menor = numero 
    except:
        print("Voce nao digitou um valor numerico. Tente novamente")
        exit()

    while numero > 0:
        if numero == 0:
            break
        
        if maior < numero:
            maior = numero
        elif menor > numero:
            menor = numero
    
        numero = float(input("Digite um valor decimal diferente de zero\n"))    

    print("O maior numero eh {:.1f} e o menor numero {:.1f}".format(maior,menor))
if __name__ == "__main__":
    main()

File: ex_lista/src_java/PontoFlutuante.java
/* Ana Cristina Silva de Oliveira - NUSP 11965630
   Turma: BCC - A 
   Exercício 06 Numero ponto flutuante
*/

import java.io.IOException;


public class PontoFlutuante {
    public static void main(String[] args) throws IOException{
        float numero = 0, maior = 0, menor = 0;

        System.out.println("Digite um valor decimal diferente de zero: ");
        numero = EntradaTeclado.leFloat();
        maior = numero;
        menor = numero;

        while (numero > 0) {
            System.out.println("Digite outro valor decimal diferente de zero: ");
            
            if (numero == 0)
                break;
            
            if (maior < numero) {
                maior = numero;
            } else if (menor > numero){
                menor = numero;
            }
            
            numero = EntradaTeclado.leFloat();
        }

        System.out.printf("O maior numero eh o %.1f e o menor numero eh o %.1f\n", maior, menor);

    }
}
